### PR TITLE
Set firebase debug on debug, adhoc but not in release

### PR DESCRIPTION
**Pull request by VincentCATILLON**
## Detailed purpose of the PR

Following this [documentation](https://firebase.google.com/docs/analytics/debugview), this PR enables debug mode for Firebase SDK in some build configurations (added a missing intermediate configuration named `Adhoc`) which is the same than Release except the config file.

### Result and observation

**iOS**:

You should be able to debug firebase analytics hits in DebugView using Adhoc artifact.

**Android**:

Any Android build can be turned into debug mode using adb:

`adb shell setprop debug.firebase.analytics.app com.coorpacademy.app`

- [ ] **Breaking changes ?**
- [ ] **Extra dependency ?**

### Testing Strategy

- [ ] Already covered by tests
- [x] Manual testing
- [ ] Unit testing

### CODE REVIEW

File: ios/fastlane/Fastfile [hunk 1]
```diff
@@ -44,7 +44,8 @@ platform :ios do
     )
     update_project_team
     gym(
-      export_method: 'ad-hoc'
+      export_method: 'ad-hoc',
+      configuration: 'Adhoc'
```
Comment (VincentCATILLON) [inline ios/fastlane/Fastfile]: This information should be interesting for you @AdrieanKhisbe

File: ios/config/release.xcconfig [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#include "../Pods/Target Support Files/Pods-Coorpacademy/Pods-Coorpacademy.release.xcconfig"
+FIREBASE_DEBUG_ENABLED = NO
```
Comment (VincentCATILLON) [inline ios/config/release.xcconfig]: This is the way we are defining config values for each build mode.

File: ios/Coorpacademy/Info.plist [hunk 3]
```diff
@@ -83,5 +85,7 @@
 	<false />
 	<key>NSMicrophoneUsageDescription</key>
 	<string>Some description</string>
+	<key>FirebaseDebugEnabled</key>
+	<string>$(FIREBASE_DEBUG_ENABLED)</string>
```
Comment (VincentCATILLON) [inline ios/Coorpacademy/Info.plist]: This is the standard way to make a config value reachable from code.

File: ios/Coorpacademy/AppDelegate.m [hunk 4]
```diff
@@ -18,7 +18,26 @@ @implementation AppDelegate
...
 {
+  NSString *plistPath = [[NSBundle mainBundle] pathForResource:@"Info" ofType:@"plist"];
+  NSDictionary *plistConfig = [[NSDictionary alloc] initWithContentsOfFile:plistPath];
+
+  // Firebase
+  BOOL isFirebaseDebugEnabled = [[plistConfig valueForKey:@"FirebaseDebugEnabled"] boolValue];
+  NSLog(@"Firebase debug %s.", isFirebaseDebugEnabled ? "enabled" : "disabled");
+
+  NSMutableArray *newArguments = [NSMutableArray arrayWithArray:[[NSProcessInfo processInfo] arguments]];
+  if (isFirebaseDebugEnabled) {
```
Comment (VincentCATILLON) [inline ios/Coorpacademy/AppDelegate.m]: The statement to enable/disable debug mode.

File: ios/Coorpacademy.xcodeproj/xcshareddata/xcschemes/Coorpacademy.xcscheme [hunk 5]
```diff
@@ -99,16 +99,6 @@
...
-      <CommandLineArguments>
-         <CommandLineArgument
-            argument = "-FIRAnalyticsDebugEnabled"
-            isEnabled = "YES">
-         </CommandLineArgument>
-         <CommandLineArgument
-            argument = "-FIRDebugEnabled"
-            isEnabled = "YES">
-         </CommandLineArgument>
-      </CommandLineArguments>
```
Comment (VincentCATILLON) [inline ios/Coorpacademy.xcodeproj/xcshareddata/xcschemes/Coorpacademy.xcscheme]: These arguments was only defined in `Run` mode in Xcode. Duplicate from build mode.

**Comment by VincentCATILLON** [comment]
Functional is OK :)

File: ios/fastlane/Fastfile (hunk 1 above)
Comment (AdrieanKhisbe) [reply ios/fastlane/Fastfile]: 👌

**Pull request merged**